bulk_indexer.go

```diff
@@ -351,10 +351,19 @@ func (b *BulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 	b.bytesUncompFlushed = bytesUncompFlushed
 	var resp BulkIndexerResponseStat
 	if res.IsError() {
-		if res.StatusCode == http.StatusTooManyRequests {
-			return resp, errorTooManyRequests{res: res}
+		e := errorFlushFailed{resp: res.String(), statusCode: res.StatusCode}
+		if res.StatusCode >= 400 {
```

Comment:
nit: can we have the switch case here as well? I find it more readable than nested if. Here we'll get the added benefit of fewer return statements.